demos/https/iot_demo_https_s3_download_async.c

```diff
@@ -774,12 +780,17 @@ static void _readReadyCallback( void * pPrivData,
             }
             else
             {
+                IotLogInfo( "Successfully reconnected to the S3 server." );
+
                 /* For all requests from the pool that have been scheduled. We want to free them so they will get
-                 * rescheduled by the main application. */
-                _freeAllScheduledRequests();
+                 * rescheduled by the main application. When a request from the pool is set to unused its associated
```

Comment:
The requests are being rescheduled because there was a reconnect?

Reply:
Any requests that are still in the queue to be sent, but haven't yet sent. These need to be rescheduled. Reconnecting clears the internal request and response queue.